src/util/exception_utils.h

```diff
@@ -33,4 +33,23 @@ class invalid_user_input_exceptiont
   std::string what() const noexcept;
 };
 
+class system_exceptiont
```

Comment:
See comment in another PR: this needs documentation.

Reply:
Could this derive from `cprover_exception_baset` to that it can be caught easily at top-level?